pkg/apis/core/validation/utils.go

```diff
@@ -567,9 +567,19 @@ func ValidateObjectReferenceNameAndNamespace(ref corev1.ObjectReference, fldPath
 	if len(ref.Name) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "must provide a name"))
 	}
+	if len(ref.Name) > 0 {
+		for _, err := range validation.IsDNS1123Subdomain(ref.Name) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("name"), ref.Name, err))
+		}
+	}
 	if requireNamespace && len(ref.Namespace) == 0 {
 		allErrs = append(allErrs, field.Required(fldPath.Child("namespace"), "must provide a namespace"))
 	}
+	if len(ref.Namespace) > 0 {
+		for _, err := range validation.IsDNS1123Subdomain(ref.Namespace) {
```

Comment:
You could use [`apivalidation.ValidateNamespaceName`](https://github.com/kubernetes/kubernetes/blob/v1.34.0/staging/src/k8s.io/apimachinery/pkg/api/validation/generic.go#L60-L63).